### PR TITLE
[SofaMisc] Add a data repository at start-up time

### PR DESCRIPTION
**Pull request by fredroy**
A small utility object to add a repo at runtime.
Check the example for getting an idea on how to use.

I guess it would be mainly useful for people still using XML scene...

______________________________________________________

This PR: 
- [x] builds with SUCCESS for all platforms on the CI.
- [x] does not generate new warnings.
- [x] does not generate new unit test failures.
- [x] does not generate new scene test failures.
- [x] does not break API compatibility.
- [x] is more than 1 week old (or has fast-merge label).

**Reviewers will merge only if all these checks are true.**

### CODE REVIEW

File: modules/SofaMisc/AddResourceRepository.cpp [hunk 1]
```diff
@@ -0,0 +1,81 @@
+#define SOFA_COMPONENT_MISC_ADDRESOURCEREPOSITORY_CPP
```
Comment (guparan) [inline modules/SofaMisc/AddResourceRepository.cpp:22]: Don't forget the license header here...

File: modules/SofaMisc/SofaMisc_test/AddResourceRepository_test.cpp [hunk 2]
```diff
@@ -0,0 +1,71 @@
+#include <SofaTest/Sofa_test.h>
```
Comment (guparan) [inline modules/SofaMisc/SofaMisc_test/AddResourceRepository_test.cpp:22]: ... and here :)

**Review by guparan — CHANGES_REQUESTED** [review]

**Review by guparan — APPROVED** [review]

**Comment by damienmarchal** [comment]
[ci-build]

**Comment by damienmarchal** [comment]
Thank for adding new features to Sofa and having provided tests with it.
Seem to compile and does not turn the CI in full red.
So let's merge it.

**Pull request merged**